src/python_framework_v02/troute/nhd_network_utilities_v02.py

```diff
@@ -445,7 +445,16 @@ def build_connections(supernetwork_parameters):
         gages = build_gages(param_df[["gages"]])
         param_df = param_df.drop("gages", axis=1)
 
-    connections = nhd_network.extract_connections(param_df, "downstream")
+    # There can be an externally determined terminal code -- that's this first value
+    terminal_codes = set()
+    terminal_codes.add(terminal_code)
+    # ... but there may also be off-domain nodes that are not explicitly identified
+    # but which are terminal (i.e., off-domain) as a result of a mask or some other
+    # an interior domain truncation that results in a 
+    # otherwise valid node value being pointed to, but which is masked out or 
+    # being intentionally separated into another domain.
+    terminal_codes = terminal_codes | set(param_df[~param_df["downstream"].isin(param_df.index)]["downstream"].values)
+    connections = nhd_network.extract_connections(param_df, "downstream", terminal_codes)
```

Comment:
I try to pass keyword arguments explicitly.
`extract_connections(param_df, "downstream", terminal_codes=terminal_codes)`